public/kbn_sankey_vis_controller.js

```diff
@@ -43,13 +44,10 @@ module.controller('KbnSankeyVisController', function ($scope, $element, $rootSco
   };
 
   let _buildVis = function (data) {
-    data.slices = filterNodesAndLinks(data.slices.nodes, data.slices.links);
```

Comment:
Please check what this function does and if its call can be fixed instead of being omitted.

Reply:
When an autoresize is done , the value of 'data.slices' should not be reinitialised again, as It contains the data formatted from the query to Elasticsearch, and this was what caused the data loss after every autoresize change.
Now I adapted the code to deal with this scenario by checking if 'data.slices' has already been initialised or not.

Reply:
Okay, if I understand you correctly the `filterNodesAndLinks` was already called when resizing, and calling it again here will produce data loss. That's fine so far.

But I don't really understand the logic. `filterNodesAndLinks(data.slices.nodes, data.slices.links);` is only called if `data.slices` evaluates to false. But wouldn't that throw an error when `data.slices.nodes` is called? Maybe it's never gonna called?!

Instead, we can maybe just move `data.slices = filterNodesAndLinks(data.slices.nodes, data.slices.links);` after line 159? Could that work?

Reply:
I made an adjustment to the code in a way that the 'filterNodesAndLinks' function call will not occurs when there is a resize event.